packaging/ubuntu-16.04/snapd.postrm

```diff
@@ -24,11 +24,14 @@ systemctl_stop() {
 }
 
 if [ "$1" = "purge" ]; then
-    # undo any bind mount to /snap that resulted from LP:#1668659
-    # (that bug can't happen in trusty -- and doing this would mess up snap.mount.service there)
-    if grep -q "/snap /snap" /proc/self/mountinfo; then
-        umount -l /snap || true
-    fi
+    # undo any bind mounts to /snap and /var/snap that resulted from parallel
+    # installs for classic snaps or LP:#1668659 (for /snap only, that bug can't
+    # happen in trusty -- and doing this would mess up snap.mount.service there)
+    for mp in /snap /var/snap; do
+        if grep -q "$mp $mp" /proc/self/mountinfo; then
```

Comment:
`grep -q " $mp $mp"` (ie, space before the first $mp) to make sure we don't catch something like `/blah/snap /snap`.